client/image_reference.go

```diff
@@ -0,0 +1,27 @@
+package client
```

Comment:
should this be in `types`?

Should we have a more general package under `types/reference`? We've talked about this several times, maybe now it's the right moment to do it